src/init.cpp

```diff
@@ -877,6 +878,11 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         InitWarning(_("Option '-upnp' is set but UPnP support was dropped in version 29.0. Consider using '-natpmp' instead."));
     }
 
+    // We removed checkpoints but keep the option to warn users who still have it in their config.
+    if (args.IsArgSet("-checkpoints")) {
+        InitWarning(_("Option '-checkpoints' is set but checkpoints were removed. This option has no effect."));
```

Comment:
This is a nice improvement. I've been running my node with `-nocheckpoints=1` for years, so now it will just throw a warning rather than abort launch. (which I tested)